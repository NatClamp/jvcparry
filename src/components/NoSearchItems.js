import React from 'react';
import { Text, Row, Col, Container, Anchor, Input, Icon, Button } from "atomize";

const NoProducts = (props) => {
  const { searchValue, resetSearch, handleChange, handleSubmit } = props;
  return (
    <>
      <Container>
        <Row d='flex' flexDir={{ xs: 'column', md: 'row' }} justify={{ xs: 'center', md: 'space-around' }} p={{ t: '2rem' }}>
          <Col size={{ xs: '12', md: '5' }}>
            <Text tag="p" textSize="title">DMsGuild products</Text>
          </Col>
          <Col size={{ xs: '12', md: '2' }} d='flex' justify='center'></Col>
          <Col size={{ xs: '12', md: '5' }} d='flex' align='center' justify={{ xs: 'center', md: 'flex-end' }}  >
            <form style={{ width: '85%' }} onSubmit={handleSubmit}>
              <Input
                placeholder="Search"
                onChange={handleChange}
                value={searchValue}
                suffix={
                  <Icon
                    name="Search"
                    size="20px"
                    cursor="pointer"
                    onClick={handleSubmit}
                    pos="absolute"
                    top="50%"
                    right="1rem"
                    transform="translateY(-50%)"
                  />
                }
              />
            </form>
            {<Button d='flex' align='center' m={{ x: '1rem' }} p={{ x: '0.75rem' }} onClick={resetSearch} bg='white' >
              <Icon name="Close" color="black" size="20px" />
              <Text textSize='caption' textColor='black' p={{ x: '0.5rem' }}>Clear</Text>
            </Button>}

          </Col>
        </Row>
        <Row d='flex' flexDir='column' justify='center' align='center'>
          <Text textSize='title' p='2rem' textAlign='center'>Sorry, there are no products matching the search term "{searchValue}"</Text>
          <Anchor href='/products/dmsguild'>
            <Button
              prefix={
                <Icon
                  name="Back"
                  size="16px"
                  color="white"
                  m={{ r: "0.5rem" }}
                />
              }
              shadow="3"
              hoverShadow="4"

            >
              Back to products
  </Button>
          </Anchor>
        </Row>
        <Row p={{ y: '2rem', l: '2rem' }}>
          <Col size={{ xs: '12', md: '7' }} d='flex' align='center'></Col>
          <Col size={{ xs: '12', md: '5' }} d='flex' justify='center' ></Col>
        </Row>
      </Container>
    </>
  )
}

export default NoProducts