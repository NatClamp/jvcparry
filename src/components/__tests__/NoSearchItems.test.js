/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import NoSearchItems from '../NoSearchItems';

describe('NoSearchItems component', () => {
  afterEach(cleanup);
  const history = createMemoryHistory();
  it('should render the NoSearchItems component', () => {
    render(
      <Router history={history}>
        <NoSearchItems searchValue="" resetSearch={() => {}} handleChange={() => {}} handleSubmit={() => {}} />
      </Router>,
    );
    expect(screen.getByText(/Sorry, there are no products matching the search term/i)).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <NoSearchItems searchValue="" resetSearch={() => {}} handleChange={() => {}} handleSubmit={() => {}} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
