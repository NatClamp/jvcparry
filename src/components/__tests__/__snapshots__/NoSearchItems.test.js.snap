// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSearchItems component should match the snapshot 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <div
      className=""
    >
      <p
        className=""
      >
        DMsGuild products
      </p>
    </div>
    <div
      className=""
    />
    <div
      className=""
    >
      <form
        onSubmit={[Function]}
        style={
          Object {
            "width": "85%",
          }
        }
      >
        <div
          className=""
        >
          <input
            className=""
            d="flex"
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
          <svg
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M11,16 C13.7614237,16 16,13.7614237 16,11 C16,8.23857625 13.7614237,6 11,6 C8.23857625,6 6,8.23857625 6,11 C6,13.7614237 8.23857625,16 11,16 Z M16.6063847,15.1921711 L19.7071068,18.2928932 C20.0976311,18.6834175 20.0976311,19.3165825 19.7071068,19.7071068 C19.3165825,20.0976311 18.6834175,20.0976311 18.2928932,19.7071068 L15.1921711,16.6063847 C14.0235906,17.4815965 12.5723351,18 11,18 C7.13400675,18 4,14.8659932 4,11 C4,7.13400675 7.13400675,4 11,4 C14.8659932,4 18,7.13400675 18,11 C18,12.5723351 17.4815965,14.0235906 16.6063847,15.1921711 Z"
              fill="#000000"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </form>
      <button
        className=""
        onClick={[Function]}
      >
        <svg
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M13.4142136,12 L15.7071068,9.70710678 C16.0976311,9.31658249 16.0976311,8.68341751 15.7071068,8.29289322 C15.3165825,7.90236893 14.6834175,7.90236893 14.2928932,8.29289322 L12,10.5857864 L9.70710678,8.29289322 C9.31658249,7.90236893 8.68341751,7.90236893 8.29289322,8.29289322 C7.90236893,8.68341751 7.90236893,9.31658249 8.29289322,9.70710678 L10.5857864,12 L8.29289322,14.2928932 C7.90236893,14.6834175 7.90236893,15.3165825 8.29289322,15.7071068 C8.68341751,16.0976311 9.31658249,16.0976311 9.70710678,15.7071068 L12,13.4142136 L14.2928932,15.7071068 C14.6834175,16.0976311 15.3165825,16.0976311 15.7071068,15.7071068 C16.0976311,15.3165825 16.0976311,14.6834175 15.7071068,14.2928932 L13.4142136,12 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
            fill="#000000"
            fillRule="evenodd"
          />
        </svg>
        <p
          className=""
        >
          Clear
        </p>
      </button>
    </div>
  </div>
  <div
    className=""
  >
    <p
      className=""
    >
      Sorry, there are no products matching the search term "
      
      "
    </p>
    <a
      className=""
      href="/products/dmsguild"
    >
      <button
        className=""
      >
        <svg
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12.2925,15.4625 C12.6825,15.8525 12.6725,16.4925 12.2925,16.8825 C11.9025,17.2725 11.2725,17.2725 10.8825,16.8825 L6.2925,12.2925 C5.9025,11.9025 5.9025,11.2725 6.2925,10.8825 L10.8825,6.2925 C11.2725,5.9025 11.9025,5.9025 12.2925,6.2925 C12.6825,6.6825 12.6825,7.3125 12.2925,7.7025 L8.4125,11.5825 L12.2925,15.4625 Z M18.2925,15.4625 C18.6825,15.8525 18.6725,16.4925 18.2925,16.8825 C17.9025,17.2725 17.2725,17.2725 16.8825,16.8825 L12.2925,12.2925 C11.9025,11.9025 11.9025,11.2725 12.2925,10.8825 L16.8825,6.2925 C17.2725,5.9025 17.9025,5.9025 18.2925,6.2925 C18.6825,6.6825 18.6825,7.3125 18.2925,7.7025 L14.4125,11.5825 L18.2925,15.4625 Z"
            fill="#FFFFFF"
            fillRule="evenodd"
          />
        </svg>
        Back to products
      </button>
    </a>
  </div>
  <div
    className=""
  >
    <div
      className=""
    />
    <div
      className=""
    />
  </div>
</div>
`;
